File: src/main/java/io/bindernews/thegrackle/cards/SummonEgrets.java
package io.bindernews.thegrackle.cards;

import com.megacrit.cardcrawl.actions.unique.WhirlwindAction;
import com.megacrit.cardcrawl.cards.DamageInfo;
import com.megacrit.cardcrawl.characters.AbstractPlayer;
import com.megacrit.cardcrawl.monsters.AbstractMonster;
import io.bindernews.bnsts.CardVariables;
import io.bindernews.thegrackle.cardmods.ExtraHitsMod;

public class SummonEgrets extends BaseCard {
    public static final CardConfig C =
            new CardConfig("SummonEgrets", CardType.ATTACK, CardRarity.UNCOMMON, CardTarget.ALL_ENEMY);
    static final CardVariables VARS = CardVariables.config(c -> {
        c.cost(-1);
        c.damage(5, 8);
        c.addModifier(ExtraHitsMod::new);
        c.multiDamage(true, true);
    });

    public SummonEgrets() {
        super(C, VARS);
    }

    @Override
    public void use(AbstractPlayer p, AbstractMonster m) {
        addToBot(new WhirlwindAction(p, multiDamage, DamageInfo.DamageType.NORMAL, freeToPlayOnce, -1));
    }
}

File: src/main/kotlin/net/bindernews/grackle/cards/SummonEgrets.kt
package net.bindernews.grackle.cards

import com.megacrit.cardcrawl.actions.unique.WhirlwindAction
import com.megacrit.cardcrawl.cards.DamageInfo
import com.megacrit.cardcrawl.characters.AbstractPlayer
import com.megacrit.cardcrawl.monsters.AbstractMonster
import net.bindernews.grackle.cardmods.ExtraHitsMod
import net.bindernews.grackle.helper.CardVariables

class SummonEgrets : BaseCard(C, VARS) {
    override fun use(p: AbstractPlayer, m: AbstractMonster) {
        addToBot(WhirlwindAction(p, multiDamage, DamageInfo.DamageType.NORMAL, freeToPlayOnce, -1))
    }

    companion object {
        @JvmField val C = CardConfig("SummonEgrets", CardType.ATTACK, CardRarity.UNCOMMON, CardTarget.ALL_ENEMY)
        val VARS = CardVariables.config { c: CardVariables ->
            c.cost(-1)
            c.damage(5, 8)
            c.addModifier(ExtraHitsMod())
            c.multiDamage(true, true)
        }
    }
}